src/Spout/Reader/Helper/XLSX/SharedStringsHelper.php

```diff
@@ -34,6 +34,9 @@ class SharedStringsHelper
     /** Value to use to escape the line feed character ("\n") */
     const ESCAPED_LINE_FEED_CHARACTER = '_x000A_';
 
+    /** This will increase your memory usage but can improve your execution time */
+    static public $KEEP_ALL_STRINGS_IN_MEMORY = false;
```

Comment:
Yes indeed. My problem is that this is VERY slow and grows exponential. In my case I tried to read the file with 130k shared strings and 60k rows. When I just read the first 1k rows spout needs ~8 seconds to read this, 5k rows = ~8sec but with 7k rows spout needs 30seconds. To read the complete file I needed more than 360seconds (I interrupted at that point maybe it would much longer maybe not...). I totally unterstand your problem with this 'fix'. This is just a fast way around in cost of memory. But the caching need to be done much more effective. In the worst case the shared strings helper read every cell a new cache file because the shared strings aren't really ordered by their position.